modules/standard/IO.chpl

```diff
@@ -6972,20 +6974,43 @@ proc fileWriter.styleElement(element:int):int {
   Iterate over all of the lines ending in ``\n`` in a :record:`fileReader` - the
   fileReader lock will be held while iterating over the lines.
 
-  Only serial iteration is supported. This iterator will halt on internal
-  system errors.
+  This iterator will halt on internal system errors.
+
+  **Example:**
+
+  .. code-block:: chapel
+
+    var r = openReader("ints.txt"),
+        sum = 0;
+
+    forall line r.lines(targetLocales=Locales) with (+ reduce sum) {
+      sum += line:int
+    }
 
   .. warning::
 
-    This iterator executes on the current locale. This may impact multilocale
-    performance if the current locale is not the same locale on which the
-    fileReader was created.
+    For serial iteration, this iterator executes on the current locale. This
+    may impact multilocale performance if the current locale is not the same
+    locale on which the fileReader was created.
 
-  :arg stripNewline: Whether to strip the trailing ``\n`` from the line. Defaults to false
+  .. warning::
+
+    For parallel iteration, this procedure does not support specifying multiple
+    locales when reading from a memory-file (e.g. files opened with
+    :proc:`openMemFile`)
+
+  :arg stripNewline: Whether to strip the trailing ``\n`` from the line —
+                     defaults to false
+  :arg targetLocales: The locales on which to read the lines (parallel
+                      iteration only) — defaults to the fileReader's home
+                      locale
   :yields: lines from the fileReader, by default with a trailing ``\n``
 
- */
-iter fileReader.lines(stripNewline = false) {
+*/
+iter fileReader.lines(
+  stripNewline = false,
+  targetLocales: [] locale = [here,]
+) {
```

Comment:
Should this match the other and declare the yielded type `: string`?

Somewhat relatedly, I would imagine we want to be able to read the lines in parallel as `bytes`. Do we have in mind an optional argument or something for that?

Reply:
We could definitely add an optional argument to specify a `string`/`bytes` return type. Do you think that belongs in this PR or separate?

Reply:
IMO it could go either way. This PR is adding `targetLocales` as an optional argument.

BTW we probably need to run the proposed new interface by a few more people to review (for module stability). I think it makes sense to cover `type yieldType=string` and `targetLocales` at the same time.

Reply:
Sounds good, I added the return type option. I'll bring this up the design in the Monday meeting to get team approval.